src/main/kotlin/graphics/scenery/volumes/vdi/VDINode.kt

```diff
@@ -0,0 +1,177 @@
+import graphics.scenery.ShaderProperty
+import graphics.scenery.backends.Shaders
+import graphics.scenery.compute.ComputeMetadata
+import graphics.scenery.compute.InvocationType
+import graphics.scenery.textures.Texture
+import graphics.scenery.utils.Image
+import net.imglib2.type.numeric.integer.UnsignedIntType
+import net.imglib2.type.numeric.real.FloatType
+import org.joml.Matrix4f
+import org.joml.Vector3f
+import org.joml.Vector3i
+import org.lwjgl.system.MemoryUtil
+import java.nio.ByteBuffer
+
+class VDINode(windowWidth: Int, windowHeight: Int, val numSupersegments: Int, vdiData: VDIData) : RichNode() {
+
+    @ShaderProperty
+    var ProjectionOriginal = Matrix4f()
+
+    @ShaderProperty
+    var invProjectionOriginal = Matrix4f()
+
+    @ShaderProperty
+    var ViewOriginal = Matrix4f()
+
+    @ShaderProperty
+    var invViewOriginal = Matrix4f()
+
+    @ShaderProperty
+    var ViewOriginal2 = Matrix4f()
+
+    @ShaderProperty
+    var invViewOriginal2 = Matrix4f()
+
+    @ShaderProperty
+    var useSecondBuffer = false
+
+    @ShaderProperty
+    var invModel = Matrix4f()
+
+    @ShaderProperty
+    var volumeDims = Vector3f()
+
+    @ShaderProperty
+    var nw = 0f
+
+    @ShaderProperty
+    var vdiWidth: Int = 0
+
+    @ShaderProperty
+    var vdiHeight: Int = 0
+
+    @ShaderProperty
+    var totalGeneratedSupsegs: Int = 0
+
+    @ShaderProperty
+    var do_subsample = false
+
+    @ShaderProperty
+    var max_samples = 50
+
+    @ShaderProperty
+    var sampling_factor = 0.1f
+
+    @ShaderProperty
+    var downImage = 1f
+
+    @ShaderProperty
+    var skip_empty = true
+
+    @ShaderProperty
+    var stratified_downsampling = false
+
+    @ShaderProperty
+    var printData = true
+
+    private val vulkanProjectionFix =
+        Matrix4f(
+            1.0f, 0.0f, 0.0f, 0.0f,
+            0.0f, -1.0f, 0.0f, 0.0f,
+            0.0f, 0.0f, 0.5f, 0.0f,
+            0.0f, 0.0f, 0.5f, 1.0f)
+
+    fun Matrix4f.applyVulkanCoordinateSystem(): Matrix4f {
+        val m = Matrix4f(vulkanProjectionFix)
+        m.mul(this)
+
+        return m
+    }
+
+    enum class DoubleBuffer {
+        First,
+        Second
+    }
+
+    var numGridCells = Vector3f()
+
+    init {
+        name = "vdi node"
+        setMaterial(ShaderMaterial(Shaders.ShadersFromFiles(arrayOf("AmanatidesJumps.comp"), this@VDINode::class.java)))
+
+        val opBuffer = MemoryUtil.memCalloc(windowWidth * windowHeight * 4)
+
+        material().textures["OutputViewport"] = Texture.fromImage(
+            Image(opBuffer, windowWidth, windowHeight),
+            usage = hashSetOf(Texture.UsageType.LoadStoreImage, Texture.UsageType.Texture))
+        metadata["ComputeMetadata"] = ComputeMetadata(
+            workSizes = Vector3i(windowWidth, windowHeight, 1),
+            invocationType = InvocationType.Permanent
+        )
+        visible = true
+
+        updateMetadata(vdiData)
+    }
+
+    fun setAccelGridSize(size: Vector3i? = null) {
```

Comment:
Please no abbreviations in function names

Reply:
Also, the `size` parameter given to the function is not used at all.

Reply:
Fixed in 555cd8026ec722c7be3295f0c8f82ea0698f259e